File: practice_set/QuickSort.java
import java.util.*;

public class QuickSort{

    public static void qs(int[] arr, int low, int high){
        if(low < high){
            int partition_index = sortPivot(arr,low,high);
            qs(arr,low,partition_index - 1);
            qs(arr,partition_index+1,high);

        }
    }

    public static int sortPivot(int[] arr,int low, int high){
        int pivot = arr[low];
        int i = low;
        int  j = high;

        while(i < j){
            while(arr[i] <= pivot&& i <= high - 1){
                i++;
            }
            while(arr[j] >pivot && j >= low+1){
                j--;
            }
            if(i < j){
                int t = arr[i];
                arr[i] = arr[j];
                arr[j] = t;
            }
        }
        int t = arr[low];
        arr[low] = arr[j];
        arr[j] = t;
        return j;
    }
   

    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);

        int[] arr = new int[5];
        System.out.println("Enter the elements in the array: ");
        for(int i=0;i<5;i++){
            arr[i] = sc.nextInt();
        }

        qs(arr,0,arr.length - 1);
        for(int i =0;i<5;i++){
            System.out.print(arr[i] + " ");
        }


    }
    }

File: ds_lab_assignments/search-sort/quickSort.cpp
#include <iostream>
using namespace std;

int partition(int arr[],int low,int high){
    int p = arr[low];
    int i = low + 1;
    int j = high;

    while (i < j)
    {
        while(p >= arr[i]){
            i++;
        }

        while(p < arr[j]){
            j--;
        }

        if(i<j){
            int t = arr[i];
            arr[i] = arr[j];
            arr[j] = t;
        }
    }

    int t = arr[low];
    arr[low] = arr[j];
    arr[j] = t;

    return j;
    
}


void quickSort(int arr[], int low, int high){
    if (low < high)
    {
        int pivot = partition(arr,low,high);

        quickSort(arr,low,pivot-1);
        quickSort(arr,pivot + 1,high);
    }
    
}
int main(){
    int n;
    cout<<"Enter the size of array : ";
    cin>>n;

    int arr[n];
    cout<<"Enter the "<<n<<" elements: "<<endl;
    for(int i = 0; i < n; i++){
        cin>>arr[i];
    }

    quickSort(arr,0,n-1);

    cout<<"Sorted array: ";
    for(int i = 0; i < n ; i++){
        cout<<arr[i]<<" ";
    }
    return 0;
}